### PR TITLE
feat(AutoMod): add support for `mention_raid_protection_enabled`

### PR DESCRIPTION
**Pull request by Victorsitou**
## Summary

Implements https://github.com/discord/discord-api-docs/pull/5778.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/automod.py [hunk 1]
```diff
@@ -296,7 +302,7 @@ def __init__(
         ...
 
     @overload
-    def __init__(self, *, mention_total_limit: int) -> None:
+    def __init__(self, *, mention_total_limit: int, mention_raid_protection_enabled: bool) -> None:
```
Comment (shiftinv) [inline disnake/automod.py:305]: `mention_raid_protection_enabled` here should have some sort of default value, probably `False` or `None`?
The runtime default will always be `None`, but the overload should have a default value too.

Not sending it when creating a rule results in the API setting it to `false`; on the other hand, it's set to `true` by default in new guilds.
(apparently it's some sort of builtin rule with a hardcoded ID (`1030554520465440818`) now? it's a bit weird.)

Comment (Victorsitou) [reply disnake/automod.py:305]: > `mention_raid_protection_enabled` here should have some sort of default value, probably `False` or `None`?

Hmm, if we do that, then we should set a default to `mention_total_limit` too.

Comment (shiftinv) [reply disnake/automod.py:305]: why?
It'd be similar to keyword rules, where `keyword_filter` is required (no default), and `allow_list` is optional (defaulting to `None`).

Comment (Victorsitou) [reply disnake/automod.py:305]: Now that I think about it, you're right, we should set a default value. But should it be true or false? 🤔

Comment (Victorsitou) [reply disnake/automod.py:305]: ~~bump~~, I guess it should be false (although it's true when the rule is deleted)

Comment (shiftinv) [reply disnake/automod.py:305]: > I guess it should be false
yeah, makes sense.

Comment (Victorsitou) [reply disnake/automod.py]: https://github.com/DisnakeDev/disnake/pull/898/commits/6a402e35cd0c2abfea50453c7d6e8e788250285c 😁

**Review by shiftinv — APPROVED** [review]
lgtm, thanks

~ci failures are unrelated

**Pull request merged**